### PR TITLE
dragonball: add metrics support for legacy device

### PR DESCRIPTION
**Pull request by lisongqian**
This PR unifies the legacy device metrics interface and outputs the metrics to runtime.

Fixes: #7248

Signed-off-by: lisongqian <mail@lisongqian.cn>

### CODE REVIEW

**Comment by lisongqian** [comment]
Block on #7505.

File: src/dragonball/src/device_manager/legacy.rs [hunk 1]
```diff
@@ -69,15 +69,19 @@ impl LegacyDeviceManager {
 #[cfg(target_arch = "x86_64")]
 pub(crate) mod x86_64 {
     use super::*;
+    use crate::metric::METRICS;
```
Comment (ZizhengBian) [inline src/dragonball/src/device_manager/legacy.rs:72]: clear up the use statement

File: src/dragonball/src/device_manager/legacy.rs [hunk 2]
```diff
@@ -69,15 +69,19 @@ impl LegacyDeviceManager {
...
     use dbs_device::device_manager::IoManager;
     use dbs_device::resources::Resource;
     use dbs_legacy_devices::{EventFdTrigger, I8042Device, I8042DeviceMetrics};
     use kvm_ioctls::VmFd;
 
+    pub(crate) const COM1: &str = "com1";
```
Comment (ZizhengBian) [inline src/dragonball/src/device_manager/legacy.rs:78]: `COM1_NAME`

same below

File: src/dragonball/src/device_manager/legacy.rs [hunk 3]
```diff
@@ -144,6 +162,15 @@ pub(crate) mod x86_64 {
             Ok((device, eventfd))
         }
     }
+
+    impl Drop for LegacyDeviceManager {
+        fn drop(&mut self) {
+            // todo move METIRCS oprations to remove_device?
```
Comment (ZizhengBian) [inline src/dragonball/src/device_manager/legacy.rs:168]: pls finish todo

File: src/dragonball/src/metric.rs [hunk 4]
```diff
@@ -55,4 +63,12 @@ pub struct DragonballMetrics {
     pub seccomp: SeccompMetrics,
     /// Metrics related to signals.
     pub signals: SignalMetrics,
+    /// Metrics related to i8032 device.
+    #[cfg(target_arch = "x86_64")]
+    pub i8042: HashMap<String, Arc<I8042DeviceMetrics>>,
```
Comment (ZizhengBian) [inline src/dragonball/src/metric.rs:68]: we only have one i8042 device and one rtc device, so I think we don't need a hashmap here.

File: src/dragonball/src/device_manager/legacy.rs [hunk 5]
```diff
@@ -160,8 +179,6 @@ pub(crate) mod aarch64 {
     pub const COM1: &str = "com1";
```
Comment (ZizhengBian) [inline src/dragonball/src/device_manager/legacy.rs:179]: please align the name with x86

**Review by ZizhengBian — APPROVED** [review]
LGTM

File: src/dragonball/src/device_manager/legacy.rs (hunk 5 above)
Comment (lisongqian) [reply src/dragonball/src/device_manager/legacy.rs]: Fixed.

**Review by quanweiZhou — APPROVED** [review]

**Review by studychao — APPROVED** [review]

File: src/dragonball/src/device_manager/legacy.rs [hunk 6]
```diff
@@ -86,14 +90,22 @@ pub(crate) mod x86_64 {
...
 
             let exit_evt = EventFd::new(libc::EFD_NONBLOCK).map_err(Error::EventFd)?;
-            let i8042_device = Arc::new(Mutex::new(I8042Device::new(
-                EventFdTrigger::new(exit_evt.try_clone().map_err(Error::EventFd)?),
-                Arc::new(I8042DeviceMetrics::default()),
-            )));
+            let i8042_device = Arc::new(Mutex::new(I8042Device::new(EventFdTrigger::new(
+                exit_evt.try_clone().map_err(Error::EventFd)?,
+            ))));
+            METRICS.write().unwrap().i8042 = i8042_device.lock().unwrap().metrics();
```
Comment (studychao) [inline src/dragonball/src/device_manager/legacy.rs:108]: seem like there is no registeration for `rtc_pl031` device?

Comment (lisongqian) [reply src/dragonball/src/device_manager/legacy.rs:108]: The `rtc_pl031` device is only registered under the `aarch64`. It works on line 198.

**Pull request merged**